tests/queries/0_stateless/02668_parse_datetime_in_joda_syntax.sql

```diff
@@ -321,7 +321,35 @@ select parseDateTime64InJodaSyntaxOrZero('2024-10-09 10:30:10.123', 'yyyy-dd-MM
 select parseDateTime64InJodaSyntaxOrZero('wrong value', 'yyyy-dd-MM HH:mm:ss.SSS') = toDateTime64('1970-01-01 00:00:00.000', 3);
 select parseDateTime64InJodaSyntaxOrZero('2023-02-29 11:22:33America/Los_Angeles', 'yyyy-MM-dd HH:mm:ssz') = toDateTime64('1970-01-01 00:00:00', 0);
 select parseDateTime64InJodaSyntaxOrZero('', '') = toDateTime64('1970-01-01 00:00:00', 0);
-select parseDateTime64InJodaSyntaxOrZero('2177-10-09 10:30:10.123', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('1970-01-01 00:00:00.000', 3);
+select parseDateTime64InJodaSyntaxOrZero('2177-10-09 10:30:10.123', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('2177-10-09 10:30:10.123', 3);
+
+-- Test with inputs between 1900 - 2299 using function parseDateTime64InJodaSyntaxOrZero
+select parseDateTime64InJodaSyntaxOrZero('1923-05-15 14:45:30.123', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('1923-05-15 14:45:30.123', 3);
+select parseDateTime64InJodaSyntaxOrZero('2150-11-23 08:22:10.456', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('2150-11-23 08:22:10.456', 3);
+select parseDateTime64InJodaSyntaxOrZero('2107-07-19 19:30:45.789123', 'yyyy-MM-dd HH:mm:ss.SSSSSS') = toDateTime64('2107-07-19 19:30:45.789123', 6);
+select parseDateTime64InJodaSyntaxOrZero('1909-12-31 23:59:59.999', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('1909-12-31 23:59:59.999', 3);
+select parseDateTime64InJodaSyntaxOrZero('2200-01-01 00:00:00.000', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('2200-01-01 00:00:00.000', 3);
+select parseDateTime64InJodaSyntaxOrZero('1924-10-09 10:30:10.123456America/Los_Angeles', 'yyyy-MM-dd HH:mm:ss.SSSSSSz') = toDateTime64('1924-10-09 10:30:10.123456', 6, 'America/Los_Angeles');
+select parseDateTime64InJodaSyntaxOrZero('2299-10-09 10:30:10.123456America/Los_Angeles', 'yyyy-MM-dd HH:mm:ss.SSSSSSz') = toDateTime64('2299-10-09 17:30:10.123456', 6);
+
+-- Test with inputs between 1900 - 2299 using function parseDateTime64InJodaSyntax
+select parseDateTime64InJodaSyntax('1923-05-15 14:45:30.123', 'yyyy-MM-dd HH:mm:ss.SSS') = toDateTime64('1923-05-15 14:45:30.123', 3);
```

Comment:
The choice of test cases is too messy. To make this more systematic, please check
- one string smaller than the lower border of DateTime64, e.g. `1899-12-31 11:22:33`,
- one string equal to the lower border of DateTime64: `1900-01-01 00:00:00`,
- one string bigger than the lower border of DateTime64 but smaller than the lower border of DateTime, e.g. `1910-05-06 11:22:33`,
- as well as equivalent test cases for the upper border of DateTime64.